src/pal/inc/pal.h

```diff
@@ -139,12 +139,24 @@ extern "C" {
 #endif
 
 #define MAX_PATH 260
-#define _MAX_PATH   260 /* max. length of full pathname */
+#define _MAX_PATH 260
 #define _MAX_DRIVE  3   /* max. length of drive component */
 #define _MAX_DIR    256 /* max. length of path component */
 #define _MAX_FNAME  256 /* max. length of file name component */
 #define _MAX_EXT    256 /* max. length of extension component */
 
+// In some Win32 APIs MAX_PATH is used for file names (even though 256 is the normal file system limit)
+// use _MAX_PATH_FNAME to indicate these cases
+#define MAX_PATH_FNAME MAX_PATH
+
+#if defined(PATH_MAX)
+#define MAX_LONGPATH   PATH_MAX  /* max. length of full pathname */
+#elif defined(MAXPATHLEN)
+#define MAX_LONGPATH   MAXPATHLEN
+#else
+#define MAX_LONGPATH   1024
+#endif
```

Comment:
If we used 4096, and then this ran, for example, on OS X which has a lower bound, what would happen if someone tried to use a path of, say, length 2000?  Since this is just an intermediate step that's just meant to unblock more cases while the switch to using the dynamic value happens, I'd suggest playing it say and going with 1024.